FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -5,38 +5,80 @@
 import CoreData
 
 public final class CoreDataFeedStore: FeedStore {
-	private static let modelName = "FeedStore"
+	private static let modelName = "CoreDataFeedStore"
+
 	private static let model = NSManagedObjectModel(name: modelName, in: Bundle(for: CoreDataFeedStore.self))
-	
+
 	private let container: NSPersistentContainer
 	private let context: NSManagedObjectContext
-	
+
 	struct ModelNotFound: Error {
 		let modelName: String
 	}
-	
+
 	public init(storeURL: URL) throws {
 		guard let model = CoreDataFeedStore.model else {
 			throw ModelNotFound(modelName: CoreDataFeedStore.modelName)
 		}
-		
+
 		container = try NSPersistentContainer.load(
 			name: CoreDataFeedStore.modelName,
 			model: model,
 			url: storeURL
 		)
 		context = container.newBackgroundContext()
 	}
-	
+
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		context.perform {
+			do {
+				if
+					let cache = try CoreDataFeedCache.fetch(context: self.context) {
+					let feed = self.mapToLocalFeed(feedItems: cache.feedItems)
+					completion(.found(feed: feed, timestamp: cache.timestamp))
```

Comment:
Following the convention in your Core Data models, the `mapToLocalFeed` should be an extension of the cache model (e.g., `localFeed` computed var).

For example:

```suggestion
				if let cache = try CoreDataFeedCache.fetch(context: self.context) {
					completion(.found(feed: cache.localFeed, timestamp: cache.timestamp))
```